fili-core/src/main/java/com/yahoo/bard/webservice/data/PartialDataHandler.java

```diff
@@ -27,6 +29,52 @@
 
     /**
      * Find the holes in the passed in intervals at a given granularity.
+     *
+     * @param constraint  Constraint containing all the column names the request depends on
+     * @param physicalTables  the tables whose column availabilities are checked
+     * @param requestedIntervals  The intervals that may not be fully satisfied
+     * @param granularity  The granularity at which to find missing intervals
+     *
+     * @return subintervals of the requested intervals with incomplete data
+     */
+    public SimplifiedIntervalList findMissingTimeGrainIntervals(
+            DataSourceConstraint constraint,
+            Set<PhysicalTable> physicalTables,
+            @NotNull SimplifiedIntervalList requestedIntervals,
+            Granularity granularity
+    ) {
+        return findMissingTimeGrainIntervals(
+                physicalTables.stream().map(table -> table.withConstraint(constraint)),
+                requestedIntervals,
+                granularity
+        );
+    }
+
+
+    /**
+     * Find the holes in the passed in intervals at a given granularity.
+     *
+     * @param constrainedTableStream  the stream of tables whose column availabilities are checked
+     * @param requestedIntervals  The intervals that may not be fully satisfied
+     * @param granularity  The granularity at which to find missing intervals
+     *
+     * @return subintervals of the requested intervals with incomplete data
+     */
+    public SimplifiedIntervalList findMissingTimeGrainIntervals(
```

Comment:
It's used in PartialDataRequestHandler.
There are two entry points to this partial data code, before table selection (and constraint) and after.
This is the entry point for 'after'